packages/firestore/src/local/local_store.ts

```diff
@@ -1130,3 +1197,131 @@ export async function ignoreIfPrimaryLeaseLoss(
     throw err;
   }
 }
+
+/**
+ * Applies the documents from a bundle to the "ground-state" (remote)
+ * documents.
+ *
+ * LocalDocuments are re-calculated if there are remaining mutations in the
+ * queue.
+ */
+export function applyBundleDocuments(
+  localStore: LocalStore,
+  documents: BundledDocuments
+): Promise<MaybeDocumentMap> {
+  const bundleConverter = new BundleConverter(localStore.serializer);
```

Comment:
If you follow the Datastore pattern, you would start this off with:

```
const localStoreImpl = debugCast(localStore, LocalStoreImpl);
```

Otherwise, this functions are tree-shakeable as is. Thanks!

Reply:
Will do once we have addressed inheritance issue in `master`.